drafts/draft-0002.md

```diff
@@ -0,0 +1,121 @@
+3. **Secure** - The specification should not expose secret keys in any form. It should include requirements on how wallets can keep their users secure by ensuring that the user has enough visibility into the transaction that she is about to sign.
+4. **Future-Proof** - The specification and its implementation should be designed in a way that allows operations that may be added to the Stellar Network in the future to be easily incorporated into the proposed specification. This will ensure that developers need to make minimal upgrades to their applications and wallets to support additions to the network protocol. The specification should also leave room for expansion to accommodate future use cases that were not anticipated at the time of writing.
+
+## Specification
+A standardized URI Scheme can achieve the goals laid out above. The syntax of the scheme will contain all the information needed to make the necessary payment to the application’s account.
+
+The scheme name will be `stellar` and should always be followed by a colon and two forward slashes to look like this: `stellar://`. The syntax for the URI will look like this: `stellar://<operation>/?<param1>=<value1>&<param2>=<value2>`, where `operation`, and the options for the query params are defined below.
+
+### Operation `tx`
+The `tx` operation represents a request to sign a specific XDR `Transaction`. The parameters for the `tx` op are as follows:
+1. `xdr` (required) -  A Stellar transaction in XDR format that is base64 encoded and then URL-encoded. If the source account in the xdr is all zeros then the URI handler should replace the source account and sequence number with the user's source account and sequence number before it signs it. If the source account is set and the sequence number is 0 then the URI handler should replace only the sequence number before signing. 
+2. `submit` (optional) - If this value is omitted then the signed XDR should be submitted to the network. If the value is present then it should be interpreted as a URL-encoded callback. The URL-encoded callback will be prefixed with its own namespace to denote whether this is a `url` callback or some other form of callback. In the case where it is a url callback (denoted by `url:`) the URI handler should send the signed XDR to this url in a `POST` request with `Content-Type` set to `application/x-www-form-urlencoded` with the data fields `xdr` containing the signed XDR (URL-encoded). If there are any query params specified in the URL callback then those should be included in the URL when submitting. For now only `url` callback types are supported.
```

Comment:
Should it actually be defined what the behavior is if submit is not defined? Why is the default to submit to the network? I know it says _should_ -- but is that actually a requirement?

Consider a multisig account -- the application handling the signing request should be the one deciding how to coordinate the rest of the signatures if submit is not present. In its current state it seems like they would be required to submit a transaction that may not have the proper signatures. The note on multisig wallets seems to conflict with this parameter.

To provide a real world use case with StellarGuard:

1) Application submits request for payment.
2) Wallet handles the URI and builds and signs the transaction with one signature.
3) Wallet checks if it is a StellarGuard-protected multisig account and submits it to StellarGuard for authorization.
4) Post-authorization, StellarGuard can either sign and submit it to the network, or return the transaction as another URI for further handling.

Step 3 would fall apart if submit was not present and the wallet were required to submit it to the network.

It seems like this is probably covered by the multisig note (meaning does not need to be submitted), but likely just needs some clarifying wording around this part.

Reply:
The URI handler can really do whatever it wants. The application is just giving the handler the info that it has that is presumably the user's intention. If the URI handler is managing a multi-sig account than yeah it would need to gather all the signatures before submitting to the network. I'll reword a bit to try to clarify

Reply:
Understood, and the change to "callback" improves the clarity. My concern was with the language "If this value is omitted then the signed XDR should be submitted to the network." which is not true in the multisig case (and maybe not even true in the normal case -- it's probably up to the wallet/user to decide to submit it or not).